### PR TITLE
users/userId/{userId} to users/?id=abc #966

### PR DESCRIPTION
**Pull request by endeavourmonk**
The api route `users/userId/{userId}` changed to `users/?id=abc`, along with this all the places where this api is used.

### CODE REVIEW

File: routes/users.js [hunk 1]
```diff
@@ -9,23 +9,22 @@ const { upload } = require("../utils/multer");
 const { getUserBadges } = require("../controllers/badges");
 
 router.post("/verify", authenticate, users.verifyUser);
-router.get("/userId/:userId", users.getUserById);
+router.get("/users/:userId", users.getUserById);
```
Comment (vikhyat187) [inline routes/users.js:12]: why did we add `/users` this is already present in the `index.js ` file as /users is the base url. @endeavourmonk

File: routes/users.js [hunk 2]
```diff
@@ -9,23 +9,22 @@ const { upload } = require("../utils/multer");
...
 router.get("/isUsernameAvailable/:username", authenticate, users.getUsernameAvailabilty);
-router.get("/chaincode", authenticate, users.generateChaincode);
-router.get("/:username", users.getUser);
-router.get("/:userId/intro", authenticate, authorizeRoles([SUPERUSER]), users.getUserIntro);
+router.get("/users/:username", users.getUser);
+router.get("/users/:userId/intro", authenticate, authorizeRoles([SUPERUSER]), users.getUserIntro);
 router.put("/self/intro", authenticate, userValidator.validateJoinData, users.addUserIntro);
-router.get("/:id/skills", users.getUserSkills);
-router.get("/:id/badges", getUserBadges);
+router.get("/users/:id/skills", users.getUserSkills);
```
Comment (vikhyat187) [inline routes/users.js:21]: this need not be changed

File: routes/users.js [hunk 3]
```diff
@@ -9,23 +9,22 @@ const { upload } = require("../utils/multer");
...
+router.get("/users/:id/badges", getUserBadges);
 
 // upload.single('profile') -> multer inmemory storage of file for type multipart/form-data
 router.post("/picture", authenticate, upload.single("profile"), users.postUserPicture);
 router.patch("/profileURL", authenticate, userValidator.updateProfileURL, users.profileURL);
 router.patch("/rejectDiff", authenticate, authorizeRoles([SUPERUSER]), users.rejectProfileDiff);
-router.patch("/:userId", authenticate, authorizeRoles([SUPERUSER]), users.updateUser);
-router.get("/suggestedUsers/:skillId", authenticate, authorizeRoles([SUPERUSER]), users.getSuggestedUsers);
+router.patch("/users/:userId", authenticate, authorizeRoles([SUPERUSER]), users.updateUser);
+router.get("/users/suggestedUsers/:skillId", authenticate, authorizeRoles([SUPERUSER]), users.getSuggestedUsers);
```
Comment (vikhyat187) [inline routes/users.js:29]: need not change this

**Review by vikhyat187 — CHANGES_REQUESTED** [review]

**Review by endeavourmonk — COMMENTED** [review]
users/userId/{userId} to users/?id=abc

**Comment by vikhyat187** [comment]
LGTM!

**Review by vikhyat187 — DISMISSED** [review]

**Comment by heyrandhir** [comment]
> The api route users/userId/{userId} changed to users/{userId}, along with this all the places where this api is used.

Is the description correct ? I think its query params not path params.

**Comment by heyrandhir** [comment]
tests are failing. can you please check @endeavourmonk

File: controllers/users.js
Comment (heyrandhir) [inline controllers/users.js:1]: why have you removed constants everywhere and hardcoding the messages ?

**Review by heyrandhir — CHANGES_REQUESTED** [review]

File: controllers/users.js
Comment (endeavourmonk) [reply controllers/users.js:1]: Updated to hardcoded to the constants

File: routes/users.js [hunk 4]
```diff
@@ -9,7 +9,7 @@ const { upload } = require("../utils/multer");
 const { getUserBadges } = require("../controllers/badges");
 
 router.post("/verify", authenticate, users.verifyUser);
-router.get("/userId/:userId", users.getUserById);
+router.get("/", users.getUserById);
 router.patch("/self", authenticate, userValidator.updateUser, users.updateSelf);
 router.get("/", authenticate, userValidator.getUsers, users.getUsers);
```
Comment (prakashchoudhary07) [inline routes/users.js:14]: Will `/users` API work as expected after this change?

**Review by prakashchoudhary07 — CHANGES_REQUESTED** [review]

**Comment by vikhyat187** [comment]
> tests are failing. can you please check @endeavourmonk

I think @endeavourmonk has updated wrong test, he has changed https://github.com/Real-Dev-Squad/website-backend/pull/991
The test for `/users/id` this API, but this doesn't exist right? correct me if I'm wrong.

Ideally he should make change in this https://github.com/Real-Dev-Squad/website-backend/blame/6919649f1761009a06a72254e0d2734e57b94d07/test/integration/users.test.js#LL465

cc : @endeavourmonk @prakashchoudhary07

**Pull request closed without merging**